### ISSUE TITLE
Better error message when EntityManager can't be produced for injection

### ISSUE THREAD

**Issue by Sanne** [quarkusio/quarkus#5525]
**Describe the bug**
Follow up on #5521 : 

when there are no entities, we automatically disable the Hibernate ORM extension, this implies that code attempting to use an injected EntityManager will get an error as the EntityManager producer is not generated either.

```
[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type javax.persistence.EntityManager and qualifiers [@Default]
```

We should find a way to produce a better error message to explain this particular `UnsatisfiedResolutionException`.

@mkouba could you suggest how this could be done? Does it make sense?

**Comment by jmartisk** [comment]
Do we want to forbid the whole application from running in this case? If not, one option would be that the processor can log a warning and include a mock implementation of `EntityManager`  that can be injected, but whose all methods throw a reasonable exception.
Not sure what you guys think about letting the application work in this case - but to me it kinda makes sense, at least with devmode, the developer's workflow could possibly be that they first inject an EM and only then they add an entity.

**Comment by mkouba** [comment]
So one thing you could do now is to inject the `BeanRegistrationPhaseBuildItem` in a build step and analyze all injection points obtained from `BeanRegistrationPhaseBuildItem.getContext().get(BuildExtension.Key.INJECTION_POINTS)` and if there's an injection point with required type `EntityManager` then throw a specific exception. Note that injection points are resolved/validated in the next phase - `ArcProcessor.validate()`.

**Comment by Sanne** [comment]
Took me a while, but I agree with @jmartisk - would be nice to just create a dumb `EntityManager` which throws a clear error message when/if it's actually used. We could use what @mkouba mentions to log a warning, so to make sure people are aware of the "work in progress" state and remember to clean it up.

But relates to #7148 as well cc/ @maxandersen

**Comment by maxandersen** [comment]
Yes, I want to have access to if not EntityManager than at least StatelessSession to allow for lightweight data access without requiring yet another database framework.

**Comment by famod** [comment]
We just had to add a Dummy entity to one of our "persistence support" Maven modules which is using sequences, to make the `QuarkusTest` work. This is a bit awkward.
Maybe it would come in handy to enable an extension via a annotation (on the test class)?

**Comment by yrodiere** [comment]
Closing in favor of #7148
